hardhat/scripts/deploy/00-mock-yearn/03-mock-yvyfi-deploy.ts

```diff
@@ -0,0 +1,42 @@
+import { ethers } from 'hardhat';
+import { DeployFunction } from 'hardhat-deploy/types';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+import { TEST_ENV } from '../../../constants';
+import { MockRegistry__factory } from '../../../typechain';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const useProxy = !hre.network.live;
+  if (TEST_ENV) return !useProxy;
+
+  const { deployer } = await hre.getNamedAccounts();
+  const { deploy } = hre.deployments;
+
+  const signer = await ethers.getSigner(deployer);
+  const yRegistry = MockRegistry__factory.connect(
+    (await hre.deployments.get('MockRegistry')).address,
+    signer
+  );
+
+  const yfi = await deploy('YFI', {
+    contract: 'MockToken',
+    from: deployer,
+    args: ['yearn.finance', 'YFI'],
+    log: true,
+  });
+
+  // Note: We aren't using the MockVaultFactory.createVault() method here because
+  // deploymentInfo won't track any transactions that create new contracts.
+  // So, we are manually deploying the Vault and registering it with the MockRegistry.
+  const yvYFI = await deploy('yvYFI', {
+    contract: 'MockVault',
+    from: deployer,
+    args: [yfi.address, 'YFI yVault', 'yvYFI'],
+  });
```

Comment:
These deploy scripts aren't in alignment with the engineering principle of `Don't Repeat Yourself`